### PR TITLE
48767_Subject_Repository

### PR DESCRIPTION
**Pull request by Omid07**

### CODE REVIEW

File: app/Http/Controllers/Supervisor/SubjectsController.php [hunk 1]
```diff
@@ -83,8 +89,9 @@ public function edit($id)
      */
     public function update(Request $request, $id)
     {
-        CourseSubject::whereSubjectId($id)->whereCourseId($request->course_id)->update(['is_finished' => 1, 'end_date' => date('Y-m-d h:i:sa')]);
-        $subject = subject::findOrFail($id);
+        // CourseSubject::whereSubjectId($id)->whereCourseId($request->course_id)->update(['is_finished' => 1, 'end_date' => date('Y-m-d h:i:sa')]);
+        $this->subjects->findSubject($id)->findCourse($request->course_id)->update(['is_finished' => 1, 'end_date' => date('Y-m-d h:i:sa')]);
```
Comment (Omid07) [inline app/Http/Controllers/Supervisor/SubjectsController.php]: here i am trying replace the comment line.. but i am getting error..i searched for it nut did not get any specific ans.. how can i call two function in one line ?? if i use $this->subjects->findSubject($id)->$this->subjects->findCourse().. it also give me a error..

Comment (dinhhoanglong91) [inline app/Http/Controllers/Supervisor/SubjectsController.php]: `$this->subjects->findSubject($id)` returns `QueryBuilder::get()` so you get a collection, not Repository. Therefore, you cannot continue searching.
There are several way to solve this problem, but, the easiest one is to move all into Repository
```php
// CourseSubjectRepository
public function updateBySubjectIdAndCourseId($subjectId, $courseId, $courseSubject)
{
    return CourseSubject::whereSubjectId($subjectId)
    ->whereCourseId($courseId)
    ->update($courseSubject)
}
// call
$this->courseSubjectRepository->updateByCourseIdAndSubjectId($id, $request->course_id, ['is_finished' => 1, 'end_date' => date('Y-m-d H:i:s')];

File: app/Repositories/SubjectRepository.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+    public function findSubject($id)
+    {
+        return Subject::findOrFail($id);
+    }
+    public function createSubject($input)
+    {
+        return Subject::create($input)->id;
+    }
+    public function updateByCourseIdAndSubjectId($subjectId, $courseId, $courseSubject)
```
Comment (dinhhoanglong91) [inline app/Repositories/SubjectRepository.php]: This function should be written in `CourseSubjectRepository`, not `SubjectRepository`

**Comment by dinhhoanglong91** [comment]
ok

**Pull request merged**